### PR TITLE
New engineering objectives

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Addss two engineering objectives, keep the sm up and keep the power up

## Why It's Good For The Game
Engineering SEVERELY lacks crew objectives, only having one for standard engineers and a unique one for the ce. The standard engie one is also extremely easy to complete, usually being accomplished by accident. These new objectives encourage engineers to DO THEIR JOB (power and not creating a singulo)

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image

</details>

## Changelog
:cl:
add: New engineer crew objectives
/:cl:

### CODE REVIEW

File: code/modules/crew_objectives/engineering_objectives.dm [hunk 1]
```diff
@@ -32,3 +32,41 @@
...
+/datum/objective/crew/apc
+	explanation_text = "Make sure the station has above (something broke message us)kW in the powernet."
+	jobs = "chiefengineer,stationengineer,atmospherictechnician"
+
+/datum/objective/crew/apc/New()
+	. = ..()
+	target_amount = rand(100,500)  // no i dont know how much a normal amount is :>
+	explanation_text = "Make sure the station has above [target_amount]kW in the powernet."
+
+/datum/objective/crew/apcc/check_completion()
```
Comment (Penwin0) [inline code/modules/crew_objectives/engineering_objectives.dm]:
```suggestion
/datum/objective/crew/apc/check_completion()
```

File: code/modules/crew_objectives/engineering_objectives.dm [hunk 2]
```diff
@@ -32,3 +32,41 @@
...
+	return ..()
+
+
+/datum/objective/crew/apc
+	explanation_text = "Make sure the station has above (something broke message us)kW in the powernet."
+	jobs = "chiefengineer,stationengineer,atmospherictechnician"
+
+/datum/objective/crew/apc/New()
+	. = ..()
+	target_amount = rand(100,500)  // no i dont know how much a normal amount is :>
```
Comment (Penwin0) [inline code/modules/crew_objectives/engineering_objectives.dm:52]: Then you should probably check so this isn't either ludicrous or way too easy

File: code/modules/crew_objectives/engineering_objectives.dm [hunk 3]
```diff
@@ -32,3 +32,41 @@
...
+	var/list/powermonitors = list()
+
+	for(var/obj/machinery/computer/monitor/pMon in GLOB.machines)
+		if(pMon.stat & (NOPOWER | BROKEN)) //check to make sure the computer is functional
+			continue
+		if(pMon.is_secret_monitor) //make sure it isn't a secret one (ie located on a ruin), allowing people to metagame that the location exists
+			continue
+		powermonitors += pMon
+
+	for(var/obj/machinery/computer/monitor/pMon in powermonitors)
```
Comment (Penwin0) [inline code/modules/crew_objectives/engineering_objectives.dm]: We don't need two loops what is done here can go in the previous one, I also don't think you are doing the power checking in a proper way, make sure you get the station powernet and only add it once to the count.

Comment (etherware-novice) [reply code/modules/crew_objectives/engineering_objectives.dm]: ok whats the proper way to get the current powernet? i know very little about pnet code

Comment (Penwin0) [reply code/modules/crew_objectives/engineering_objectives.dm]: Make sure you get a power monitor connected to the station and only add up the available power once.

Comment (etherware-novice) [reply code/modules/crew_objectives/engineering_objectives.dm]: *how do i do that*

Comment (Penwin0) [reply code/modules/crew_objectives/engineering_objectives.dm]: Just check which powermonitor is connected to the powernet with the most power and use only that one I suppose

Comment (etherware-novice) [reply code/modules/crew_objectives/engineering_objectives.dm]: yea what commands do i use

Comment (Penwin0) [reply code/modules/crew_objectives/engineering_objectives.dm]: Just see which of the monitors you are looping through has the most power and check for that one

File: code/modules/crew_objectives/engineering_objectives.dm [hunk 4]
```diff
@@ -32,3 +32,41 @@
...
+		var/datum/powernet/connected_powernet = pMon.get_powernet()
+		total_power += connected_powernet.viewavail
```
Comment (PowerfulBacon) [inline code/modules/crew_objectives/engineering_objectives.dm]: This doesn't work at all, multiple power nets on the same grid will add the same powernets twice.
Just loop through all powernets in the world, its a good enough solution

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/modules/crew_objectives/engineering_objectives.dm (hunk 4 above)
Comment (etherware-novice) [reply code/modules/crew_objectives/engineering_objectives.dm]: so... just replace it with for(obj/whatever/powernet in world?)

Comment (PowerfulBacon) [reply code/modules/crew_objectives/engineering_objectives.dm]: I dont know, in world is usually pretty bad. Find a good way to get all the powernets (The power subsystem?)

**Pull request closed without merging**